gcc/jit/dummy-frontend.cc

```diff
@@ -1280,6 +1280,37 @@ recording::type* tree_type_to_jit_type (tree type)
     recording::type* element_type = tree_type_to_jit_type (inner_type);
     return element_type->get_pointer ();
   }
+  else if (type == unsigned_intTI_type_node)
+    return new recording::memento_of_get_type (&target_builtins_ctxt,
+      GCC_JIT_TYPE_UINT128_T);
+  else if (INTEGRAL_TYPE_P (type))
+  {
+    unsigned int size = tree_to_uhwi (TYPE_SIZE_UNIT (type));
+    return target_builtins_ctxt.get_int_type (size, TYPE_UNSIGNED (type));
+  }
+  else if (SCALAR_FLOAT_TYPE_P (type))
+  {
+    unsigned int size = tree_to_uhwi (TYPE_SIZE_UNIT (type));
+    enum gcc_jit_types type;
+    switch (size)
+    {
+      case 2:
+	/* FIXME: Wrong type.  Update when Float16 is supported.  */
+	type = GCC_JIT_TYPE_FLOAT;
+	break;
```

Comment:
Would it be better to assert here, rather than silently use the wrong type?

Reply:
or reuse the `default:` branch

Reply:
I can try, but the whole issue with this function as that if it asserts, libgccjit basically becomes unusable, because we now try to process all builtins at the start. Which means if we get a type we do not know how to convert, this will abort.

As @pinskia mentionned elsewhere, this approach is kinda wrong, but given the way libgccjit works (in 2 phases, without access to the target stuff in the recording phase), I do not know any better idea.
Does any of you have a better idea?

The only idea I have to workaround this situation (which is still not ideal) would be to disable this by default and ask the user to set some option to enable this so that this doesn't affect the people that don't want to use target builtins.